src/gui/mzroll/eicwidget.cpp

```diff
@@ -1727,6 +1727,16 @@ void EicWidget::setPeakGroup(PeakGroup* group) {
     emit groupSet(group);
     replot(group);
     _clearEicPoints();
+    // Done to update group in case of alignment performed
+    // after peak detection.
+    for (size_t i = 0; i < group->peaks.size(); i++) {
+        auto peak = group->peaks[i];
+        for (auto eic : eicParameters->eics) {
+            if (eic->sample == peak.getSample()) {
+                group->peaks[i].rt = eic->rt[peak.pos];
+            }
```

Comment:
This is a nice attempt to get rid of floating bubbles. But the fix is only visual. If I export one of these groups (whose peak RTs were updated after alignment), here's a snapshot of the peak-detailed report:

<img width="924" alt="Screenshot 2020-05-06 at 2 00 31 PM" src="https://user-images.githubusercontent.com/10397162/81155176-4bf45880-8fa2-11ea-854d-1e2947f66c6c.png">

A peak's `rt` should be a value between its `rtmin` and `rtmax`. I have highlighted a few cases in the screenshot, where you can see that it is not.

My point is that we should aim to actually update every attribute of every peak of every peak-group in every peak-table, just after an alignment operation. I understand all of this can be confusing to someone who has not worked on these data structures for a long time. So, let me know if you need some help with this.